src/Supero/NightfallProtocol/network/packets/PlayerArmorDamagePacket.php

```diff
@@ -0,0 +1,114 @@
+        $result->legsSlotDamage = $legsSlotDamage;
+        $result->feetSlotDamage = $feetSlotDamage;
+        $result->bodySlotDamage = $bodySlotDamage;
+        return $result;
+    }
+
+    public function getHeadSlotDamage() : ?int{ return $this->headSlotDamage; }
+
+    public function getChestSlotDamage() : ?int{ return $this->chestSlotDamage; }
+
+    public function getLegsSlotDamage() : ?int{ return $this->legsSlotDamage; }
+
+    public function getFeetSlotDamage() : ?int{ return $this->feetSlotDamage; }
+
+    public function getBodySlotDamage() : ?int{ return $this->bodySlotDamage; }
+
+    private function maybeReadDamage(int $flags, int $flag, PacketSerializer $in) : ?int{
+        if(($flags & (1 << $flag)) !== 0){
+            return $in->getVarInt();
+        }
+        return null;
+    }
+
+    protected function decodePayload(PacketSerializer $in) : void{
+        $flags = $in->getByte();
+
+        $this->headSlotDamage = $this->maybeReadDamage($flags, self::FLAG_HEAD, $in);
+        $this->chestSlotDamage = $this->maybeReadDamage($flags, self::FLAG_CHEST, $in);
+        $this->legsSlotDamage = $this->maybeReadDamage($flags, self::FLAG_LEGS, $in);
+        $this->feetSlotDamage = $this->maybeReadDamage($flags, self::FLAG_FEET, $in);
+        if($in->getProtocolId() >= CustomProtocolInfo::PROTOCOL_1_21_20){
+            $this->bodySlotDamage = $this->maybeReadDamage($flags, self::FLAG_BODY, $in);
+        }
+    }
+
+    private function composeFlag(?int $field, int $flag) : int{
+        return $field !== null ? (1 << $flag) : 0;
+    }
+
+    private function maybeWriteDamage(?int $field, PacketSerializer $out) : void{
+        if($field !== null){
+            $out->putVarInt($field);
+        }
+    }
+
+    /**
+     * @param CustomPacketSerializer $out
+     * @return void
+     */
+    protected function encodePayload(PacketSerializer $out) : void{
+        $out->putByte(
+            $this->composeFlag($this->headSlotDamage, self::FLAG_HEAD) |
+            $this->composeFlag($this->chestSlotDamage, self::FLAG_CHEST) |
+            $this->composeFlag($this->legsSlotDamage, self::FLAG_LEGS) |
+            $this->composeFlag($this->feetSlotDamage, self::FLAG_FEET) |
+            ($out->getProtocol() >= CustomProtocolInfo::PROTOCOL_1_21_20 ? $this->composeFlag($this->bodySlotDamage, self::FLAG_BODY) : 0)
+        );
+
+        $this->maybeWriteDamage($this->headSlotDamage, $out);
+        $this->maybeWriteDamage($this->chestSlotDamage, $out);
+        $this->maybeWriteDamage($this->legsSlotDamage, $out);
+        $this->maybeWriteDamage($this->feetSlotDamage, $out);
+        if($out->getProtocol() >= CustomProtocolInfo::PROTOCOL_1_21_20){
+            $this->maybeWriteDamage($this->bodySlotDamage, $out);
+        }
+    }
+
+    /**
+     * @throws ReflectionException
+     */
+    public function getConstructorArguments(PM_Packet $packet): array
+    {
+        $properties = [

```

Comment:
Again, no need for reflection

Reply:
![image](https://github.com/user-attachments/assets/0da6e1fc-c432-4d96-9531-cff903ee519d)

Reply:
It private ?

Reply:
![image](https://github.com/user-attachments/assets/11d6a97c-5364-4868-91b3-4eee4404c0ed)

Reply:
use $packet->get(Field)()

Reply:
![image](https://github.com/user-attachments/assets/48a88e4e-bb77-414b-a820-ddc687d6b438)

Reply:
Ow, sr i dont see it !! :))

Reply:
https://github.com/Nightfall-MCPE/NightfallProtocols/pull/4/commits/ded0d1db70e9d0a0b17facc4cd228310d3ac5b45